### ISSUE TITLE
Unsupported restart policy "unless-stopped"

### ISSUE THREAD

**Issue by brunohenriquy** [rancher-sandbox/rancher-desktop#3998]
### Actual Behavior

While trying to use the restart policy “unless-stopped”, I got the following message:
<img width="665" alt="Screenshot 2023-02-20 at 02 26 14" src="https://user-images.githubusercontent.com/8320519/220203141-339cedee-9cb0-4e06-9545-0ce6294ec212.png">

Doing some investigation at nerdctl code, I found this lines at **checkRestartCapabilities**:
```
        res, err := client.IntrospectionService().Plugins(ctx, []string{"id==restart"})
	if err != nil {
		return false, err
	}
	if len(res.Plugins) == 0 {
		return false, fmt.Errorf("no restart plugin found")
	}
	restartPlugin := res.Plugins[0]
	capabilities := restartPlugin.Capabilities
	if len(capabilities) == 0 {
		capabilities = []string{"always"}
	}
	if !strutil.InStringSlice(capabilities, policySlice[0]) {
		return false, fmt.Errorf("unsupported restart policy %q, supported policies are: %q", policySlice[0], restartPlugin.Capabilities)
	}
```
It seems it could’t find the policies available from containerd so it set default to “always” only, which is the only one working here.
After some research, I came across this command to list containerd plugins:
```
ctr plugins ls
```
So I tried:
```
❯ rdctl shell ctr plugins ls
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
Error: exit status 255
```
Is this a containerd problem, or rancher-desktop? Maybe because of M1 architecture.

The response of:
```
res, err := client.IntrospectionService().Plugins(ctx, []string{"id==restart"})
```
Is:
```
"plugins:{type:\"io.containerd.internal.v1\"  id:\"restart\"  requires:\"io.containerd.event.v1\"  requires:\"io.containerd.service.v1\"}" 
```

So it seems it acquired the response about the plugin “restart”, but the Capabilities where not returned.

Reference link to Slack https://rancher-users.slack.com/archives/C0200L1N1MM/p1676871987097119

### Steps to Reproduce

Use unless-stopped

### Result

<img width="665" alt="Screenshot 2023-02-20 at 02 26 14" src="https://user-images.githubusercontent.com/8320519/220203382-c4089182-cf40-4e42-bf3b-b99f314d6f01.png">

### Expected Behavior

Accept unless-stopped restart policy

### Additional Information

```
❯ nerdctl system info                                                                                                                                         
Client:
 Namespace:     default
 Debug Mode:    false

Server:
 Server Version: v1.6.8
 Storage Driver: overlayfs
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Log: fluentd journald json-file syslog
  Storage: native overlayfs
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.15.78-0-virt
 Operating System: Alpine Linux v3.16
 OSType: linux
 Architecture: aarch64
 CPUs: 2
 Total Memory: 5.786GiB
 Name: lima-rancher-desktop
 ID: 19421722-8793-4554-8547-e448d05e6100

### Rancher Desktop Version

1.7.0

### Rancher Desktop K8s Version

1.25.4

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

macOS

### Operating System / Build Version

macOS Ventura 13.2

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

_No response_

**Comment by gunamata** [comment]
@brunohenriquy , As per the comment https://github.com/containerd/nerdctl/issues/1605#issuecomment-1336927577 , this is at the containerd level, and the fix will be available in containerd v1.7 release which is not out yet. Once it is released, the containerd version in Rancher Desktop will be upgraded in one of the upcoming releases.

**Comment by brunohenriquy** [comment]
> @brunohenriquy , As per the comment [containerd/nerdctl#1605 (comment)](https://github.com/containerd/nerdctl/issues/1605#issuecomment-1336927577) , this is at the containerd level, and the fix will be available in containerd v1.7 release which is not out yet. Once it is released, the containerd version in Rancher Desktop will be upgraded in one of the upcoming releases.

Oh Thanks @gunamata!

**Comment by gunamata** [comment]
@jandubois , Reopening after our discussion about containerd 1.7.0 upgrade in the triage call.
